sonarqube/run-sonar-analysis.sh

```
@@ -0,0 +1,219 @@
+  clone_dir="$WORK_DIR/$safe_branch_name"
+  
+  # Clone the specific branch
+  echo "Cloning branch: $branch into $clone_dir"
+  if ! git clone --single-branch --branch "$branch" --depth 1 "$REPO_URL" "$clone_dir" 2>&1 | tee /tmp/clone_${safe_branch_name}.log; then
+    echo "ERROR: Failed to clone branch: $branch"
+    failed_branches+=("$branch (clone failed)")
+    continue
+  fi
+  
+  # Change to cloned directory
+  cd "$clone_dir"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)